### ISSUE TITLE
Failed to handle RPCResponseSystem

### ISSUE THREAD

**Issue by SlayksWood** [StrangeLoopGames/EcoIssues#21486]
original issue #21162, crash was fixed but we still have this exception.
> Receiving RPC Response error when using hammer to put down blocks.
> 
> ![image.png](https://camo.githubusercontent.com/050342a777129629628751e6ac53e41b1d7dc9f9adaedbc63498aa76dbab7dde/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3630313762656463626630653662666231346531323030332f61616334633433342d373336312d343466352d383138612d616265376336633436623131)
> 
> Issue happens during private server session, no mods installed or used.
> 
> Unable to provide reproduction steps as this issue seems to happen 70% of the time (will provide steps when able to reproduce issue reliably)
> 
> Error message from log as follow :
```
[05:56:20.109] Failed to handle RPCResponseSystem.NullReferenceException: Object reference not set to an instance of an object.
at HasViewTooltipNoClick.OnGotBSON (Eco.Shared.Serialization.BSONObject info) [0x00000] in <00000000000000000000000000000000>:0
at System.Action1[T].Invoke (T obj) [0x00000] in <00000000000000000000000000000000>:0
at Eco.Shared.Networking.RPCManager+<>c__DisplayClass22_01[T].b__0 (System.Object o) [0x00000] in <00000000000000000000000000000000>:0
at System.Action`1[T].Invoke (T obj) [0x00000] in <00000000000000000000000000000000>:0
at Eco.Shared.Networking.RPCManager.HandleQueryResponse (Eco.Shared.Serialization.BSONObject bson) [0x00000] in <00000000000000000000000000000000>:0
at NetworkManager.Eco.Shared.Networking.INetworkEventHandler.ReceiveEvent (Eco.Shared.Networking.INetClient client, Eco.Shared.Networking.NetworkEvent netEvent, Eco.Shared.Serialization.BSONObject bson) [0x00000] in <00000000000000000000000000000000>:0
at Eco.Shared.Networking.NetObject.ReceiveEvent (Eco.Shared.Networking.INetClient client, Eco.Shared.Networking.NetworkEvent netEvent, Eco.Shared.Serialization.BSONObject bsonObj)
```

**Comment by SlayksWood** [comment]
> Happened randomly on tool switch/movement from building to digging
> 
> ```
> [07:36:29.903] Failed to handle RPCResponseSystem.NullReferenceException: Object reference not set to an instance of an object.
>   at HasViewTooltipNoClick.OnGotBSON (Eco.Shared.Serialization.BSONObject info) [0x00000] in <00000000000000000000000000000000>:0 
>   at Steamworks.Callback`1+DispatchDelegate[T].Invoke (T param) [0x00000] in <00000000000000000000000000000000>:0 
>   at Eco.Shared.Networking.RPCManager+<>c__DisplayClass22_0`1[T].<Call>b__0 (System.Object o) [0x00000] in <00000000000000000000000000000000>:0 
>   at Steamworks.Callback`1+DispatchDelegate[T].Invoke (T param) [0x00000] in <00000000000000000000000000000000>:0 
>   at Eco.Shared.Networking.RPCManager.HandleQueryResponse (Eco.Shared.Serialization.BSONObject bson) [0x00000] in <00000000000000000000000000000000>:0 
>   at NetworkManager.Eco.Shared.Networking.INetworkEventHandler.ReceiveEvent (Eco.Shared.Networking.INetClient client, Eco.Shared.Networking.NetworkEvent netEvent, Eco.Shared.Serialization.BSONObject bson) [0x00000] in <00000000000000000000000000000000>:0 
>   at Eco.Shared.Networking.NetObject.ReceiveEvent (Eco.Shared.Networking.INetClient client, Eco.Shared.Networking.NetworkEvent netEvent, Eco.Shared.Serialization.BSONObject bsonObj) [0x00000] in <00000000000000000000000000000000>:0 
>   at ClientPacketQueueHandler.TryFetchNextClientUpdate (Eco.Shared.Time.TimeLimit timeLimit) [0x00000] in <00000000000000000000000000000000>:0 
>   at ClientPacketHandler.HandleNetworkEvents (Eco.Shared.Time.TimeLimit timeLimit) [0x00000] in <00000000000000000000000000000000>:0 
>   at System.Action`1[T].Invoke (T obj) [0x00000] in <00000000000000000000000000000000>:0 
>   at FramePlanner.PlannerGroup.OnUpdate () [0x00000] in <00000000000000000000000000000000>:0 
>   at FramePlanner.FramePlannerSystem.OnUpdate () [0x00000] in <00000000000000000000000000000000>:0 
>   at Unity.Entities.ComponentSystem.Update () [0x00000] in <00000000000000000000000000000000>:0 
>   at Unity.Entities.ComponentSystemGroup.UpdateAllSystems () [0x00000] in <00000000000000000000000000000000>:0 
>   at Unity.Entities.ComponentSystem.Update () [0x00000] in <00000000000000000000000000000000>:0 
>   at UnityEngine.Application+LowMemoryCallback.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
> ```

**Comment by SlayksWood** [comment]
> Tested on 0.9.3 Staging-1971
> 
> Happens more frequently when building. Most especially when switching between fill types.
> 
> Adding log.
> [RCP Error.txt](https://github.com/StrangeLoopGames/EcoIssues/files/6265634/RCP.Error.txt)
> 
> > [11:06:05.552] Disconnected with error: Failed to handle RPCResponse
> > EcoEngine.Logging.LogManager:LogFormat(LogType, Object, String, Object[])
> > UnityEngine.Logger:Log(LogType, Object)
> > UnityEngine.Debug:LogWarning(Object)
> > NetworkManager:OnClientDisconnected(String)
> > DG.Tweening.Core.DOSetter`1:Invoke(T)
> > Eco.Shared.Networking.NetworkClient:HandleDisconnect(String)
> > Eco.Shared.Networking.Udp.UdpDisconnectedHandler:Invoke(Object, DisconnectReason)
> > Eco.Networking.ENet.ENetUdpClient:OnDisconnected(ENetConnection, DisconnectReason)
> > Unity.Serialization.<>c:.cctor()
> > Eco.Networking.ENet.ENetUdpPeer:Service(Int32)
> > <>c__DisplayClass65_0:b__0()
> > System.Threading.ThreadStart:Invoke()
> > System.Threading.ContextCallback:Invoke(Object)
> > System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
> > System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
> > System.Threading.ThreadStart:Invoke()
> 
> Following the tutorial for building a room can also trigger this issue. The frequency is about 6 out of 10 tries.
> The most cause trigger is switching between fill types while building.

**Comment by TwixterZ** [comment]
Fix tested and confirmed as of 9.4 Staging 2009

_Tested with about 1k blocks while constantly changing its shape and fill types_
![image](https://user-images.githubusercontent.com/78054747/120919353-ba4aa380-c6eb-11eb-96f6-e9de71b1610e.png)
